components/aws/sagemaker/commonv2/sagemaker_component.py

```diff
@@ -0,0 +1,709 @@
+                return resource
+
+            sleep(period_length)
+
+        logging.error(
+            f"Wait for resource %s to be consumed by controller timed out",
+            (self.job_name),
+        )
+        return None
+
+    def _get_resource_exists(self) -> bool:
+        """Check if the custom resource exists.
+
+        Returns:
+            bool: True if the resource exists, False otherwise.
+        """
+        try:
+            return self._get_resource() is not None
+        except ApiException:
+            return False
+
+    def _create_resource(
+        self,
+        cr_spec: object,
+        wait_periods=6,
+        period_length=10,
+    ):
+        """Create a resource from the spec and wait to be consumed by
+        controller.
+
+        Args:
+            cr_spec: A dictionary object representing the custom object.
+            wait_periods: The number of times to wait for the resource to be created.
+            period_length: The length of time to wait between polling.
+        """
+
+        resource = self._create_custom_resource(cr_spec)
+        resource = self._wait_resource_consumed_by_controller(
+            wait_periods, period_length
+        )
+
+        if resource is None:
+            logging.error(f"Resource {self.job_name} is not created.")
+            logging.error(
+                f"Possible reason: ACK controller may not have been configured properly."
+            )
+            raise Exception(f"Resource {self.job_name} is not created.")
+        else:
+            logging.info(f"Created custom resource with name: {self.job_name}")
+
+        return resource
+
+    def _check_resource_conditions(self):
+        """Check the status of the custom resource.
+        * loop through all conditions
+            * if recoverable and condition set to true, print out message and return true
+            (let outside polling loop goes on forever and let user decide if should stop)
+            * if terminal and condition set up true, print out message and return false
+        """
+        status_conditions = self._get_resource()["status"]["conditions"]
+
+        for condition in status_conditions:
+            condition_type = condition["type"]
+            condition_status = condition["status"]
+            condition_message = condition.get("message", "No error message found.")
+
+            if condition_type == "ACK.Terminal" and condition_status == "True":
+                logging.error(json.dumps(condition, indent=2))
+                logging.error(
+                    "Terminating the run because resource encountered a Terminal condition. Please describe the resource for further debugging and retry with correct parameters."
+                )
+                return False
+            if condition_type == "ACK.Recoverable" and condition_status == "True":
+                logging.error(json.dumps(condition, indent=2))
+                if "ValidationException" in condition_message:
```

Comment:
Add `InvalidParameter` to this. We need to fix this in ACK but for now ok to add here

```
ERROR:root:{
  "message": "InvalidParameter: 2 validation error(s) found.\n- missing required field, CreateTrainingJobInput.VpcConfig.SecurityGroupIds.\n- missing required field, CreateTrainingJobInput.VpcConfig.Subnets.\n",
  "status": "True",
  "type": "ACK.Recoverable"
}
ERROR:root:Waiting for error to be resolved . . . Please fix the error or terminate the job and retry with correct parameters if this is not a transient error
```